src/RequestHandler/managedexports.cxx

```diff
@@ -206,18 +206,26 @@ http_read_request_bytes(
     }
     IHttpRequest *pHttpRequest = (IHttpRequest*)pInProcessHandler->QueryHttpContext()->GetRequest();
 
-    BOOL fAsync = TRUE;
-
-    hr = pHttpRequest->ReadEntityBody(
-        pvBuffer,
-        dwCbBuffer,
-        fAsync,
-        pdwBytesReceived,
-        pfCompletionPending);
 
-    if (hr == HRESULT_FROM_WIN32(ERROR_HANDLE_EOF))
+    if (pHttpRequest->GetRemainingEntityBytes() > 0)
     {
-        // We reached the end of the data
+        BOOL fAsync = TRUE;
+        hr = pHttpRequest->ReadEntityBody(
+            pvBuffer,
+            dwCbBuffer,
+            fAsync,
+            pdwBytesReceived,
+            pfCompletionPending);
+
+        if (hr == HRESULT_FROM_WIN32(ERROR_HANDLE_EOF))
+        {
+            // We reached the end of the data
+            hr = S_OK;
```

Comment:
Set hr = S_OK initially so we don't need the else. Also, don't we need to set pdwBytesReceived to 0 and pfCompletionPending to False?

Reply:
Done